app/src/main/java/com/example/drawingactivity/DrawViewModel.kt

```diff
@@ -39,10 +39,11 @@ class CustomColorViewModel : ViewModel() {
     }
 
     var layerList = mutableListOf<String>()
-    var isDialogOpen = false
-    var defaultLayerHintFound = false
-    var currentUserEnteredLayerName: String? = null
-    var layerNumber: Int = LAYER_COUNT_START
+    var isAddDialogOpen = false
+    var layerHintFound = false
+    var userNewLayerName: String? = null
```

Comment:
very confusing name for where your saving the string for interruptions like rotation, sounds more like what you are actually just adding to the list of names, maybe a different name that helps differentiate it?